runcheckendline

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+set -e
+
+no_new_line=$(find . -type f -not -path '*/\.*' \
+    -not -path "*/*egg-info/*" \
+    -exec grep -Iq . {} \; -and -print0 \
+    -not -empty | xargs -0 -L1 bash -c \
+    'test "$(tail -c 1 "$0")" && echo "$0\n" || true')
+trim_new_line=$(echo "$no_new_line" | cut -f1 -d-)
+if [[ $trim_new_line == "" ]]; then
+    echo "All check done"
```

Comment:
can you remove this line please? It's not needed, see [Art of Unix programming, rule of silence](http://www.catb.org/esr/writings/taoup/html/ch01s06.html#id2878450).

Reply:
PS: can you also rebase this branch on the current master please?

Reply:
which line i must remove? 
8/9/10/11?

Reply:
I think @nemesisdesign wants you to remove the echo at the end (line 11), referring to the website linked above. "When a program has nothing surprising to say, it should say nothing."

Reply:
correct, I'm referring to `echo "All check done"`, an inline comment on github almost always refers to the line above the comment, unless otherwise stated

Reply:
i've done fix this line and update my master branch.

Reply:
I think you still need to rebase your branch with `master` in `openwisp-utils`. If you look at your repo right now, you will see this:
![image](https://user-images.githubusercontent.com/34558138/51435671-c6348880-1c4a-11e9-94e6-bf04f6f25fe0.png)
After you are done rebasing, it should look like:
![image](https://user-images.githubusercontent.com/34558138/51435681-009e2580-1c4b-11e9-83d4-c9a429862a63.png)
If you have any questions, I'm happy to help 😃

Reply:
@ppabcd @dwang is right, you still need to update this branch with the master branch, you should be able to do that with:

```
git rebase origin/master
```